From: Shivam Kalra <shivamkalra98@zohomail.in>
Subject: mm-vmalloc-use-physical-page-count-in-vread_iter-for-vm_alloc-areas-fix
Date: Mon, 25 May 2026 16:17:47 +0530

pcpu_get_vm_areas() creates VM_ALLOC areas without setting nr_pages.
Add an nr_pages check to fall back to get_vm_area_size() for these
areas.

Link: https://lore.kernel.org/aff47da5-4fd5-481d-be18-e1eb99639490@zohomail.in
Signed-off-by: Shivam Kalra <shivamkalra98@zohomail.in>
Cc: Alice Ryhl <aliceryhl@google.com>
Cc: Danilo Krummrich <dakr@kernel.org>
Cc: Uladzislau Rezki (Sony) <urezki@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/vmalloc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/vmalloc.c~mm-vmalloc-use-physical-page-count-in-vread_iter-for-vm_alloc-areas-fix
+++ a/mm/vmalloc.c
@@ -4673,7 +4673,7 @@ long vread_iter(struct iov_iter *iter, c
 			 * the mapping without updating area->size. Other
 			 * mapping types (vmap, ioremap) don't set nr_pages.
 			 */
-			size = (vm->flags & VM_ALLOC) ?
+			size = (vm->flags & VM_ALLOC && vm->nr_pages) ?
 				       (vm->nr_pages << PAGE_SHIFT) :
 				       get_vm_area_size(vm);
 		else
_
